sklearn/decomposition/_base.py

```diff
@@ -12,12 +12,27 @@
 
 import numpy as np
 from scipy import linalg
+from scipy.sparse import issparse
+from scipy.sparse.linalg import LinearOperator
 
 from ..base import BaseEstimator, ClassNamePrefixFeaturesOutMixin, TransformerMixin
 from ..utils._array_api import _add_to_diagonal, get_namespace
 from ..utils.validation import check_is_fitted
 
 
+def _implicitly_center(X, mu):
```

Comment:
Could you document this private function?

In particular, could you briefly explain that X is (generally) sparse here and the benefit of using `LinearOperator`?

Reply:
I've written some docs, but think it could be better.